products/sentinel_one.py

```diff
@@ -414,63 +464,60 @@ def nested_process_search(self, tag: Tag, criteria: dict, base_query: dict):
             for search_field, terms in criteria.items():
                 all_terms = ', '.join(f'"{term}"' for term in terms)
 
-                if search_field not in PARAMETER_MAPPING:
+                if search_field not in self.parameter_mapping:
                     self._echo(f'Query filter {search_field} is not supported by product {self.product}',
                                logging.WARNING)
                     continue
 
-                parameter = PARAMETER_MAPPING[search_field]
+                parameter = self.parameter_mapping[search_field]
                 search_value = all_terms
 
-                if len(terms) > 1:
-                    search_value = f'({all_terms})'
-                    operator = 'in contains anycase'
-                else:
-                    operator = 'containscis'
-
                 if tag not in self._queries:
                     self._queries[tag] = list()
 
-                self._queries[tag].append(Query(from_date, to_date, parameter, operator, search_value))
+                if self._pq:
+                    for term in terms:
+                        self._queries[tag].append(Query(from_date, to_date, parameter, 'contains', f"'{term}'"))
+                else:
+                    if len(terms) > 1:
+                        search_value = f'({all_terms})'
+                        operator = 'in contains anycase'
+                    else:
+                        operator = 'containscis'
+
+                    self._queries[tag].append(Query(from_date, to_date, parameter, operator, search_value))
         except KeyboardInterrupt:
             self._echo("Caught CTRL-C. Returning what we have...")
 
-    def _process_queries(self):
-        """
-        Process all cached queries.
-        """
-        start_date = datetime.utcnow()
-        end_date = start_date
-
-        # determine earliest start date
-        for tag, queries in self._queries.items():
-            for query in queries:
-                if query.start_date < start_date:
-                    start_date = query.start_date
+    def _get_query_text(self) -> list[Tuple[Tag, str]]:
+        # tuple contains tag and full query
+        # these chunks will be combined with OR statements and executed
+        query_text = list[Tuple[Tag, str]]()
 
-        try:
-            # queries with certain operators can be combined into a more compact query format
+        if self._pq:
+            query_text = list[Tuple[Tag, str]]()
 
+            for tag, queries in self._queries.items():
+                for query in queries:
+                    full_query = f'{query.parameter} {query.operator} {query.search_value}'
+                    query_text.append((tag, full_query))
```

Comment:
When I only specify the `--query` param and don't use definition files, the query isn't built correctly

```suggestion
                    if query.full_query is not None:
                        query_text.append((tag, query.full_query))
                    else:
                        full_query = f'{query.parameter} {query.operator} {query.search_value}'
                        query_text.append((tag, full_query))
```

Error log:
```
[2023-02-23 09:14:38,187] [DEBUG   ] [surveyor.s1                         ] [common.py           :54  ] Authenticated
[2023-02-23 09:14:38,189] [DEBUG   ] [surveyor                            ] [help.py             :36  ] Running Custom Query: src.process.name contains "svchost.exe"
[2023-02-23 09:14:38,189] [DEBUG   ] [surveyor.s1                         ] [help.py             :36  ] Built Query: src.process.name contains "svchost.exe"
[2023-02-23 09:14:38,189] [DEBUG   ] [surveyor.s1                         ] [sentinel_one.py     :713 ] Entered get_results
[2023-02-23 09:14:38,189] [DEBUG   ] [surveyor.s1                         ] [sentinel_one.py     :717 ] Executing additional _process_queries
[2023-02-23 09:14:38,190] [DEBUG   ] [surveyor.s1                         ] [sentinel_one.py     :564 ] Query params: {'accountIds': ['REDACTED'], 'fromDate': 1677163478189, 'toDate': 1677165278189, 'limit': 20000, 'query': 'None None None | group count() by endpoint.name, src.process.user, src.process.image.path, src.process.cmdline, src.process.name, src.process.publisher, url.address, tgt.file.internalName, event.time, site.id, site.name'}
```